src/DependencyInjection/Configuration.php

```diff
@@ -118,6 +118,28 @@ function ($v) {
 
                     throw new InvalidConfigurationException('You need to configure a proxy_client for the logout_handler.');
                 })
+            ->end()
+            ->validate()
+                ->ifTrue(
+                    function ($v) {
+                        // Check whether 'purgekeys' has been enabled
+                        return
+                            array_key_exists('proxy_client', $v)
+                            && array_key_exists('varnish', $v['proxy_client'])
+                            && 'purgekeys' === $v['proxy_client']['varnish']['tag_mode']
+                        ;
+                    }
+                )->then(function ($v) {
+                    // Overide the header name on invalidation requests
+                    $v['proxy_client']['varnish']['tags_header'] = 'xkey-softpurge';
```

Comment:
i think we need to do those overrides conditionally, only if nothing has been set explicitly. currently, xkey-purge would not work as expected. same with the values below.

we could remove the default values and the condition above, to always apply this validation and set default values based on whether its purgekeys or not. and if a value is already set, we would not overwrite it.